yt/utilities/linear_interpolators.py

```diff
@@ -1,11 +1,82 @@
+import abc
+
 import numpy as np
 
 import yt.utilities.lib.interpolators as lib
 from yt.funcs import mylog
 
 
-class UnilinearFieldInterpolator:
-    def __init__(self, table, boundaries, field_names, truncate=False):
+class _LinearInterpolator(abc.ABC):
+    _ndim: int
+    _dim_i_type = "int32"
+
+    def __init__(self, table, field_names, truncate=False, *, store_table=True):
```

Comment:
Just a quick remark (I'll do a much more in depth review when I can): I'd put the asterisk before any boolean argument to avoid Boolean traps.

Reply:
happy to move it -- but that'd make it not backwards compatible, right? any worry there?

Reply:
(and absolutely no rush on the review)

Reply:
I didn't realize this was existing API. Indeed it'd be breaking to move the asterisk without a warning first. It's way out of scope for this PR, so ignore my nit !